k8s-operator/apis/v1alpha1/types_proxyclass.go

```diff
@@ -161,16 +161,33 @@ type Pod struct {
 	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="!(has(self.serviceMonitor) && self.serviceMonitor.enable  && !self.enable)",message="ServiceMonitor can only be enabled if metrics are enabled"
```

Comment:
It does in practice needs the 'has' else we will hit the 'no such key'. I remember reading this a while ago and wondering whether this is because we omit fields that are unset. I am not entirely sure what counts as field being defined when using this for custom resource validation

Reply:
Oh sorry, I thought I tested this out, but maybe I messed up the deploy.

Reply:
I _think_ I tested it with a custom resource like

```
kind: ProxyClass
metadata:
  name: prod
spec:
  metrics:
    enable: false
```